src/fault_injector/fault_types/track_blocked_fault.py

```diff
@@ -33,8 +34,10 @@ def resolve_fault(self, tick: int):
         :param tick: the simulation tick in which resolve_fault was called
         :type tick: Integer
         """
-        if self.track is None:
-            raise ValueError("Track not set, probably due to not injecting the fault")
+        if self.track is None or self.track is not self.get_track(
+            self.simulation_object_updater, self.track.identifier
+        ):
+            raise ValueError("Track does not exist or fault not injected")
```

Comment:
```suggestion
        if self.track is None:
            raise ValueError("Fault was not injected")
        if self.track is not self.get_track(
            self.simulation_object_updater, self.track.identifier
        ):
            raise ValueError("Track does not exist")
```
Again, this would be more specific.